wordpress-mobile/WordPress-Android
# Blaze: Show the promote with blaze card

Parent #17909 

This PR implements the Promote with Blaze entry point as a dashboard card

- Moves the PromoteWithBlazeCard to a DashboardCard so that it can be properly slotted within that group
- Moves location of ViewHolder 
- Creates the PromoteWithBlazeUpdate partial state
- Creates the builder params and the builder for the card
- Creates the PromoteWithBlazeSource and adds it the MySiteSourceManager
- Connects to the BlazeStore to get blaze status for the site
- Updates and adds new unit tests
- If the dashboard cards are in error, then the blaze card won't be shown as well

**Notes**
- There are `todo` comments in `MySiteViewModel` for implementing the clicks functionality and the tracking events. This PR was getting too big, so these tasks will be handled in a separate PR.
- Launching the blaze overlay is blocked until the overlay code is ready and is in trunk
- The **UI needs to be tidied up**, this will follow in a separate PR

Light | Dark
-- | --
<img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/219696105-aea50ff1-6538-4757-ab4b-831aa23bec8e.png"> | <img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/219696114-9ec5ed2d-a821-431a-a87a-c4f225d811b0.png">


**Merge Instructions**
This PR is dependent upon [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2659)
- Wait for the FluxC PR to be successfully merged into trunk
- Copy the trunk hash (from the PR) and apply it to build.gradle in this PR
- Push the commit
- Wait for CI to complete
- Merge as normal
**OR** - Approve the PRs only and I will merge :)

**Test**
- Install and launch Jetpack app
- Navigate to the dashboard
- ✅ Verify that the `Promote with Blaze` card is not shown
- Navigate to Me > App Settings > Debug Setttings
- Enable the `blaze` flag and restart the app
- Navigate back to the dashboard
- ✅ Verify that the `Promote with Blaze` card is shown and it's positioned after the writing prompt and/or before the post cards


## Regression Notes
1. Potential unintended areas of impact
The blaze card is shown when it shouldn't be

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and new unit tests

3. What automated tests I added (or what prevented me from doing so)
`PromoteWithBlazeCardBuilderTest` and `PromoteWithBlazeCardSourceTest`

PR submission checklist:
